packages/core/src/lib/switch.svelte

```diff
@@ -0,0 +1,136 @@
+<!--
+  @component
+  
+  The switch is used to handle a binary input.
+
+  ```svelte
+  <Switch
+    on='false'
+    label='Switch Label'
+  />
+  ```
+-->
+<svelte:options immutable />
+
+<script lang="ts">
+import cx from 'classnames';
+import { createEventDispatcher } from 'svelte';
+
+/**
+ * The switch label text.
+ */
+export let label = '';
+/**
+ * The name of the icon next to the switch label.
+ */
+export let name = '';
+/**
+ * Shows if the switch is 'on' or 'off'.
+ */
+export let on = false;
+/**
+ * Shows if the switch is annotated (if the text of 'on' or 'off' is displayed next to the switch).
+ */
+export let variant: 'annotated' | 'default' = 'default';
+/**
+ * Shows if the switch is disabled (if the disabled style of the switch is displayed).
+ */
+export let disabled = false;
+/**
+ * Shows if the switch is readonly (if the readonly style of the switch is displayed).
+ */
+export let readonly = false;
+/**
+ * Shows the message of the tooltip.
+ * TODO(APP-2290): Tooltips above icons won't work until icons are migrated.
+ */
+// export let tooltip = '';
+
+const dispatch = createEventDispatcher<{
+  /** When the switch is clicked. */
+  input: { value: boolean };
+}>();
+
+let input: HTMLInputElement;
+
+const handleClick = () => {
+  if (!(disabled || readonly)) {
+    on = !on;
+    input.checked = on;
+    dispatch('input', { value: on });
+  }
+};
+</script>
+
+<label
+  class={cx('flex w-fit flex-col justify-start gap-1', {
+    'text-disabled-dark': disabled,
+  })}
+>
+  <div class="flex items-center gap-1.5">
+    {#if label}
+      <button
+        class="w-fit text-xs"
+        on:click={handleClick}>{label}</button
+      >
+    {/if}
+
+    <!-- * TODO(APP-2290), (APP-2304): Can't specify icon for tooltip until tooltip/icons are migrated. -->
+    <!-- {#if tooltip}
+      <v-tooltip text={tooltip}>
+        <v-icon
+          class="-mt-0.5"
+          name="information-outline"
+        />
+      </v-tooltip>
+    {/if} -->
```

Comment:
Tooltip and icon are merge now, I believe!

Reply:
Thanks - will update this :)

Reply:
Updated!